### PR TITLE
fix(cli): fail closed on interactive prompts in non-TTY sessions

### PR DESCRIPTION
**Pull request by neavra**
## Problem

When stdout is piped or redirected (the headless / agent use case), the CLI treats it as the machine-output channel and auto-switches `--output` to `json`. But the interactive prompts never got the memo:

- Bare `send` / `execute` / `swap` / `tokens --add` / `join` (and the password / seedphrase prompts) still rendered an **inquirer prompt onto stdout**, corrupting the JSON channel, and then failed with a generic `UNKNOWN_ERROR` / exit 7:

  ```
  $ vultisig send ethereum 0x… 0.000001 | jq
  ? Proceed with this transaction? (y/N)      <-- ANSI prompt bytes on stdout
  {"success":false,"error":{"code":"UNKNOWN_ERROR","exitCode":7, …}}
  ```

- `completion --install` was worse — with a non-TTY stdin, tabtab's readline shell-picker closed immediately and dumped a raw `ERR_USE_AFTER_CLOSE` stack trace.

## Fix

Piped stdout now implies non-interactive, mirroring how `--output` already defaults to `json` when stdout isn't a TTY.

- `resolveNonInteractive()` folds a non-TTY stdout into the non-interactive decision in the global `preAction` hook.
- The confirm gates and `requireInteractive()` now throw a typed `ConfirmationRequiredError` with a **stable `CONFIRMATION_REQUIRED` code (exit 12)** and a clear stderr hint — *before* any prompt is drawn — instead of `UNKNOWN_ERROR`/7.
- All interactive prompts route through a shared prompt module bound to **stderr**, so they can never land on the machine-output channel.
- `completion --install` prints a graceful "needs an interactive terminal" message (with the manual-append instructions) instead of crashing.

`--yes` / `--confirm` and normal interactive TTY usage are unchanged.

## Tests

- Unit coverage for the `ConfirmationRequiredError` contract (code, exit code, JSON envelope).
- Unit coverage for `resolveNonInteractive` (TTY vs piped) and `requireInteractive` failing closed.
- `confirmTransaction` / `confirmSwap` reject with `ConfirmationRequiredError` and write **zero bytes to stdout** in non-interactive mode.
- A piped-subprocess test for `completion --install`: no prompt bytes on stdout, no `ERR_USE_AFTER_CLOSE`, exit 12, graceful stderr message.

Changeset included (`@vultisig/cli` patch).

## Summary by CodeRabbit

* **Bug Fixes**
  * CLI interactive prompting now fails closed in non-interactive/headless sessions and routes all prompt UI to **stderr** only (prevents any **stdout** leakage).
  * Added consistent `CONFIRMATION_REQUIRED` handling (`exit code 12`) with clear guidance; swap/withdraw and other confirmation-gated flows now stop before proceeding.
  * `completion --install` now refuses without an interactive terminal gracefully.
* **Documentation**
  * Updated CLI exit-code documentation to include `12` and refreshed related meanings.
* **Tests**
  * Expanded non-TTY regression coverage, including “zero stdout” assertions.
* **Chores**
  * Silenced `7z-wasm` banner/progress output on **stdout** during QR payload processing.

### CODE REVIEW

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Several prompt paths still bypass the new non-interactive guard. Commands like import/export/verify/address-book can still reach prompt() in headless runs, so the CLI does not fail closed consistently yet.

**Review by rcoderdev — APPROVED** [review]

**Review by gomesalexandre — APPROVED** [review]
nice, this now closes the original nontty fail-open gap cleanly. the shared prompt chokepoint + stderr-only prompt channel mean the prompting paths fail closed before emitting stdout bytes, and the shell/completion edge cases that bypass the preAction gate now get their own explicit tty guards.

validation:
- targeted cli tests passed: `yarn test:cli clients/cli/src/commands/nontty-prompt.test.ts clients/cli/src/core/errors.test.ts clients/cli/src/lib/output.test.ts clients/cli/src/lib/prompt.test.ts clients/cli/src/ui.test.ts` -> 72 passed
- `yarn build:shared` is still blocked by the pre-existing cardano `auxiliaryData`/`ISigningInput` type error on both `origin/main` and this branch
- `yarn workspace @vultisig/cli typecheck` and `clients/cli/src/lib/completion.test.ts` are still blocked by the unrelated pre-existing `src/commands/tx-status.ts` `isValidTxHash` import/type issue, so i did not count those against this PR

no new findings on the actual diff.

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Two must-fix regressions still leave non-interactive mode fail-open on the stdout-nonTTY path:

1. `clients/cli/src/lib/completion.ts:222`
   `completion --install` only guards on `!process.stdin.isTTY`, but this PR defines non-interactive as `!stdout.isTTY || !stdin.isTTY` in `clients/cli/src/lib/output.ts:52`. With TTY stdin and redirected stdout, `vultisig completion --install > out` can still enter the tabtab installer instead of failing closed with exit 12.

2. `clients/cli/src/commands/vault-management.ts:61` and `:760`
   the fast-vault flows still key their auto two-step / side-effect behavior off stdin only. With redirected stdout and TTY stdin, the command can create pending vault state via `createFastVault()` / `createFastVaultFromSeedphrase()` before the later verification prompt hits the shared non-interactive refusal. That is a side effect before refusal on a path this PR says should fail closed.

Please align these call sites with the shared non-interactive definition before approval.

File: clients/cli/src/lib/output.ts [hunk 1]
```diff
@@ -37,9 +39,30 @@ export function isNonInteractive(): boolean {
...
+  return explicit || !stdout.isTTY || !stdin.isTTY
+}
+
 export function requireInteractive(hint: string): void {
   if (nonInteractive) {
-    throw new Error(`Interactive prompt required but --non-interactive is set. ${hint}`)
+    throw new ConfirmationRequiredError(
+      'Interactive input required but the session is non-interactive (piped output or --non-interactive/--ci).',
+      hint
+    )
```
[the remark this answers is not in the document]
Comment (neavra) [reply clients/cli/src/lib/output.ts:65]: Fixed in `01e896f2` — the message now names both triggers: "piped/redirected stdout or stdin, or --non-interactive/--ci".

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
still not green on the fail-closed guarantee. prompt rendering itself is now guarded, but several changed command paths still write human stdout before the non-interactive refusal:

- `clients/cli/src/commands/transaction.ts:95-107` renders the preview before the `isNonInteractive()` confirmation gate.
- `clients/cli/src/commands/swap.ts:223-238` does the same with `displaySwapPreview(...)` before `confirmSwapIfNeeded()`.
- `clients/cli/src/commands/execute.ts:171-197` prints the contract preview before the non-interactive confirmation check.
- `clients/cli/src/commands/vault-management.ts:333-335` prints `Email and password are required...` before the prompt backstop on `verify --resend`.
- `clients/cli/src/commands/vault-management.ts:1129-1142` prints vault details/warnings before the delete confirmation prompt.

`info()` / `warn()` still go to stdout in `clients/cli/src/lib/output.ts`, so these are real stdout side effects on the same headless paths this PR is trying to fail closed. the guard needs to fire before any of those preview / guidance writes when interactive confirmation or prompted input is required.

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
blocking: `join secure` still violates the fail-closed contract on the stdin-nonTTY path.

- `clients/cli/src/index.ts:561-564` parses the QR, decides the session is `KEYIMPORT`, then prints `This session requires a seedphrase to join.` via `info(...)` **before** `promptSeedphrase()` hits `requireInteractive()`.
- `resolveNonInteractive()` explicitly defines non-interactive as `!stdout.isTTY || !stdin.isTTY` (`clients/cli/src/lib/output.ts:45-58`), so a piped stdin session is supposed to refuse before any stdout bytes are emitted.
- that means `join secure` can still write human stdout before the typed `CONFIRMATION_REQUIRED` refusal, which is the exact corruption bug this PR is trying to close everywhere else.

suggested fix:
- move the non-interactive guard ahead of the `info(...)` line for the `KEYIMPORT && !mnemonic` branch, or
- stop writing that guidance to stdout on this path.

coverage gap:
- the new non-TTY suite covers send / swap / execute / rujira / verify / delete, but not `join secure` with a KEYIMPORT QR that requires a mnemonic, which is why this regression is still live.

I also tried the focused CLI tests in a disposable review clone, but that clone had no installed Yarn state, so I grounded this review in the current diff + call flow rather than a green local receipt.

**Review by gomesalexandre — APPROVED** [review]
took over the last fail-closed gap and pushed fix(cli): stop verify resend from falling into nontty otp prompt on this head.

what changed
- `verify --resend --email --password` no longer resends successfully and then falls through into an impossible OTP prompt in non-interactive mode
- non-interactive resend-only now returns cleanly with follow-up guidance instead of producing a post-success `CONFIRMATION_REQUIRED`
- added a focused regression test for the resend-without-code path

receipts
- `yarn workspace @vultisig/cli vitest run src/commands/nontty-prompt.test.ts` -> 14 passed
- `yarn workspace @vultisig/cli typecheck` -> clean

I also re-read the previously flagged areas on this head: completion install, fast-vault create/import, send/swap/execute preview gating, delete preview, join secure KEYIMPORT, and the SevenZip locateFile retry. No remaining blocker found on the current diff. CI is queued on the pushed head; once it comes back green this looks good to land.
